File: 3483-alternating-groups-ii/alternating-groups-ii.py
class Solution:
    def numberOfAlternatingGroups(self, colors: list[int], k: int) -> int:
        n = len(colors)
        count = 0
        left = 0

        for right in range(1, n + k - 1):
            if colors[right % n] == colors[(right - 1) % n]:
                left = right
            if right - left + 1 >= k:
                count += 1

        return count

File: 3483-alternating-groups-ii/alternating-groups-ii.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors, int k) {
        int n = colors.size();
        int count = 0;
        int left = 0;

        for (int right = 1; right < n + k - 1; right++) {
            if (colors[right % n] == colors[(right - 1) % n]) {
                left = right;
            }
            if (right - left + 1 >= k) {
                count++;
            }
        }

        return count;
    }
};